### ISSUE TITLE
no-redeclare should ignore declare in same level but difference scope

### ISSUE THREAD

**Issue by Thaina** [eslint/eslint#6500]
I want redeclare check only if it conflict with the same or outer scope. We should stack scope check

```js
function A()
{
	for (var pid in event.Players) {
		var command = event.squad[pid];
	}

	for (var pid in data.Players) { // pid should not error
		var command = event.squad[pid];  // command should not error
		var count = 0;  // count should error here if it redeclare conflict to outer scope
	}

	var count = 0;
}
```

**Comment by mysticatea** [comment]
Thank you for this issue.

Those are redeclarations since `var` declarations are not block-scoped.
You can use `let` declarations for block-scoped variables.

Or if you strongly want to use `var` declarations with block-scoping, you can use [my plugin's `block-scoped-var` rule](https://github.com/mysticatea/eslint-plugin/blob/master/docs/rules/block-scoped-var.md) instead of `no-redeclare` rule.

**Comment by Thaina** [comment]
I see, but I still use es5 so `let` is not an option for me

Thank you very much

**Comment by ilyavolodin** [comment]
As @mysticatea mentioned, `no-redeclare` rule follows JavaScript hoisting rules. So in the example that you provide, it's doing the right thing and notifying you about redeclaration of `pid`, `command` and `count` variables.

Closing as this is working as expected.
